ddtrace/tracer/textmap.go

```diff
@@ -265,9 +301,18 @@ func (p *propagator) extractTextMap(reader TextMapReader) (ddtrace.SpanContext,
 			if err != nil {
 				return ErrSpanContextCorrupted
 			}
-			ctx.setSamplingPriority(priority)
+			ctx.setSamplingPriority("", priority, samplerNone, math.NaN())
 		case originHeader:
 			ctx.origin = v
+		case traceTagsHeader:
+			if ctx.trace == nil {
+				ctx.trace = newTrace()
+			}
+			ctx.trace.tags, err = parsePropagatableTraceTags(v)
+			ctx.trace.upstreamServices = ctx.trace.tags[keyUpstreamServices]
+			if err != nil {
+				log.Warn("did not extract trace tags (err: %s)", err.Error())
+			}
```

Comment:
`ctx.trace.upstreamServices = ctx.trace.tags[keyUpstreamServices]`

The above line will panic if `parsePropagatableTraceTags` returns a `nil` map, which it does under a number of error conditions. The error check should be done before accessing `ctx.trace.tags`.

This should ideally be covered by a test case.

Reply:
That case is covered by [this test](https://github.com/DataDog/dd-trace-go/blob/03a21337abadaf057e06dc6567ea7ff5f7678d85/ddtrace/tracer/textmap_test.go#L233).
I won't panic, because it's not a pointer but `nil` map. It simply returns an empty string.
You can check here: https://go.dev/play/p/lVFJ5HU2ofo

Reply:
Whoops, you're right. As long as we don't assign to the map we are ok.